trustmanager/keystore.go

```diff
@@ -319,35 +319,28 @@ func getSubdir(alias string) string {
 func GetPasswdDecryptBytes(passphraseRetriever notary.PassRetriever, pemBytes []byte, name, alias string) (data.PrivateKey, string, error) {
 	var (
 		passwd  string
-		retErr  error
 		privKey data.PrivateKey
 	)
 	for attempts := 0; ; attempts++ {
 		var (
 			giveup bool
 			err    error
 		)
+		if attempts > 10 {
+			return nil, "", ErrAttemptsExceeded{}
+		}
 		passwd, giveup, err = passphraseRetriever(name, alias, false, attempts)
 		// Check if the passphrase retriever got an error or if it is telling us to give up
 		if giveup || err != nil {
 			return nil, "", ErrPasswordInvalid{}
 		}
-		if attempts > 10 {
-			return nil, "", ErrAttemptsExceeded{}
-		}
 
 		// Try to convert PEM encoded bytes back to a PrivateKey using the passphrase
 		privKey, err = utils.ParsePEMPrivateKey(pemBytes, passwd)
-		if err != nil {
-			retErr = ErrPasswordInvalid{}
-		} else {
+		if err == nil {
 			// We managed to parse the PrivateKey. We've succeeded!
-			retErr = nil
 			break
 		}
 	}
-	if retErr != nil {
```

Comment:
We might want to check if privKey is null here, and if so, return some kind of error.  Alternately, in line 339, do `privKey, retErr = ...` and leave this check here.

It's possible for us to never have decrypted the private key because `utils.ParsePEMPrivateKey` could have failed over and over.